File: scripts/_smoke_test.mjs
/**
 * API smoke test — starts the Express app on a random port,
 * hits every route family, prints results, exits.
 */
import { createServer } from 'http';

process.env.VERCEL = ''; // allow the listen() branch to be skipped; we manage server ourselves

const { default: app } = await import('../server/index.js');

const server = createServer(app);

await new Promise(r => server.listen(0, '127.0.0.1', r));
const { port } = server.address();
const base = `http://127.0.0.1:${port}`;

console.log(`\n🔬 Smoke tests — server on :${port}\n`);

const tests = [
  ['GET', '/api/health'],
  ['GET', '/api/places/autocomplete?input=Austin'],
  ['GET', '/api/location/resolve?zip=78701'],
  ['GET', '/api/templates'],
  ['GET', '/api/telemetry/ping'],
  ['GET', '/api/partner/v1/health'],
  ['GET', '/api/epc/search'],
  ['GET', '/api/sales-agent/leads'],
  ['GET', '/api/opportunities'],
  ['GET', '/health'],
  // verify old sub-path catch-alls still resolve via the single top-level handler
  ['GET', '/api/quote/estimate'],
];

let passed = 0, failed = 0;

for (const [method, path] of tests) {
  try {
    const res = await fetch(base + path, { method });
    // anything < 500 is "the route exists and responded" — auth/param errors (401/400/404) are fine
    const ok = res.status < 500;
    console.log(`${ok ? '✅' : '❌'}  ${String(res.status).padEnd(4)} ${method} ${path}`);
    ok ? passed++ : failed++;
  } catch (e) {
    console.log(`❌  ERR  ${method} ${path}  (${e.message})`);
    failed++;
  }
}

console.log(`\nResult: ${passed} passed, ${failed} failed\n`);
server.close();
process.exit(failed > 0 ? 1 : 0);

File: scripts/_smoke_test.py
#!/usr/bin/env python3
"""Smoke test: verify all API routes respond through the single catch-all."""
import urllib.request, urllib.error, sys

BASE = "http://localhost:19999"
ROUTES = [
    ("health",               "/api/health"),
    ("places/autocomplete",  "/api/places/autocomplete?input=test"),
    ("location/resolve",     "/api/location/resolve?zip=90210"),
    ("templates",            "/api/templates"),
    ("telemetry",            "/api/telemetry"),
    ("quote",                "/api/quote"),
    ("sales-agent",          "/api/sales-agent"),
    ("epc",                  "/api/epc"),
    ("partner/v1/health",    "/api/partner/v1/health"),
    ("partner/auth/token",   "/api/partner/auth/token"),
    ("partner/v1/industries","/api/partner/v1/industries"),
    ("partner/v1/leads",     "/api/partner/v1/leads"),
    ("partner/v1/quotes",    "/api/partner/v1/quotes"),
    ("partner/v1/webhooks",  "/api/partner/v1/webhooks"),
]

print("=== SMOKE TEST ===")
failures = []
for label, path in ROUTES:
    try:
        with urllib.request.urlopen(BASE + path, timeout=4) as r:
            code = r.status
    except urllib.error.HTTPError as e:
        code = e.code   # 4xx/5xx are still LIVE — route exists
    except Exception as ex:
        code = 0
        failures.append(label)
    icon = "PASS" if code != 0 else "FAIL"
    print(f"  [{icon}]  HTTP {str(code):<5}  {label}")

print("==================")
if failures:
    print(f"FAILED: {failures}")
    sys.exit(1)
else:
    print("ALL ROUTES LIVE")
